topics/releases.md

```diff
@@ -1,144 +1,109 @@
+You should carefully read the release notes before performing a major version upgrade.
+Major version upgrades do not guarantee backwards compatibility, which means you should always upgrade replicas before upgrading primaries in order to ensure data consistency.
 
-### Major versions
+The Valkey community strives to make as few backwards breaking changes as possible.
+When breaking changes are required, we will also strive to provide a way to mitigate the impact without incuring downtime to your application.
 
-Major versions introduce new capabilities and significant changes.
+## Release schedule
 
-Ideally, these don't introduce application-level compatibility issues.
+The Valkey community strives to release a stable major version once a year.
+Stable minor versions are created as needed in between major releases, and we aim to release at least one minor version a year.
 
-## Release schedule
+### Release candidate
 
-A new major version is planned for release once a year.
+New minor and major versions of Valkey begin by branching off the `unstable` branch as an initial release candidate branch with a name that takes the form of `*major.minor*`, example `7.2`.
+The first release candidate, or rc1, is released once it can be used for development purposes and for testing the new version.
+Release candidate versions will start with a patch version of "0" and will take the form `*major.minor.patch-rcN*`, example `7.2.0-rc1` followed by `7.2.0-rc2`.
+At this stage, most of the new features and changes in the new version are ready for review, and the version is released for the purpose of collecting the public's feedback.
+Subsequent release candidates are released every couple of weeks, primarily for fixing bugs and refining features based off of user input.
 
-Generally, every major release is followed by a minor version after six months.
+### Stable release
 
-Patches are released as needed to fix high-urgency issues, or once a stable
-version accumulates enough fixes to justify it.
+Once development has ended and the feedback for release candidate slows down, it is ready for the final release. 
+At this point, the release is marked as stable and is released with "0" as its patch-level version.
 
-For contacting the core team on sensitive matters and security issues, please
-see [SECURITY.md](https://github.com/valkey-io/valkey/blob/unstable/SECURITY.md).
+Patches are released as needed to fix high-urgency issues, or once a stable version accumulates enough fixes to justify it.
 
 ## Support
 
-As a rule, older versions are not supported as we try very hard to make the
-API mostly backward compatible.
+The latest stable release is always fully supported and maintained.
 
-Upgrading to newer versions is the recommended approach and is usually trivial.
+The Valkey community will provide maintanence support, providing patch releases for bug fixes and all security fixes, for 3 years from when a version was first released.
```

Comment:
I think we should also clarify for which minor version we will provide "maintenance support" because below we have been very explicit that only the last minor version of each major version will receive security support.

IMO, not providing security support for other minor versions in a major version makes the "maintenance support" for non-last minor versions a moot point because users need both. So I think either we should provide both support for every minor version or align maintenance support with security support so only the last minor version of each major version is considered (my preference)

```suggestion
The Valkey community will provide maintenance support, providing patch releases for bug fixes and all security fixes, for 3 years from when a version was first released.
```

Reply:
> IMO, not providing security support for other minor versions in a major version makes the "maintenance support" for non-last minor versions a moot point because users need both.

This is definitely not true. If there is a bug in a feature you don't use, you won't see any reason to upgrade. It'll also be easier if we say we aren't going to fix it.  Many organizations are required to follow procedures for patching security issues. We have historically had about 2 security CVE fixes per year in Redis since we have so few dependencies. 

> So I think either we should provide both support for every minor version or align maintenance support with security support so only the last minor version of each major version is considered (my preference)

Everyone I've pitched this proposal to, doesn't like it. Since it will push end users to wait until LTS versions are released and then they'll upgrade, at which point, I would prefer we just launch a single major version every year with features.

Second question, what would be the support of a minor version? I don't think it should just be until the next minor is released, that is **very** uncommon.

Reply:
> If there is a bug in a feature you don't use, you won't see any reason to upgrade. 

Good point.

>If there is a security issue, you'll organization will probably make you apply it.

I read the latest revision as saying all versions (minor and major) are fully supported for up to 3 years and between year 4 and year 5, inclusive, only last minor version of each major version is supported for security. Is this understanding correct?

Here is an example, and assuming we tick-tock release every 6 months: 7.2, 8.0, 8.2, 9.0, 9.2, 10.0, 10.2, 11.0, 11.2, 12.0, and 12.2.

By the time we release 12.2, 7.2 turns 5 and 9.2 turns 3. We then have a CVE fix and a critical bug fix applicable to all these versions. With this release policy, are we saying, versions between 9.2 and 12.2, inclusive, will receive both fixes, which is 7 versions in total but 7.2 and 8.2 will receive the CVE fix only?